plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt

```diff
@@ -197,6 +200,21 @@ class Yarn2(override val descriptor: PluginDescriptor = Yarn2Factory.descriptor,
         return parsePackageInfos(process.stdout)
     }
 
+    private fun List<PackageInfo>.filterNotInstalled(installedModuleIds: Set<String>): List<PackageInfo> {
```

Comment:
Should this maybe extend `Collection`, take a `Collection` parameter, and return a `Set`?

Reply:
I want to require a `Set`, so any performance degradation of passing something different would be flagged by the compiler and for clarity. (I believe this to be consistent with the code base too)